plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFusionEnv.groovy

```diff
@@ -0,0 +1,266 @@
+    private final String endpoint
+
+    // Platform access token to use for requests
+    private final String accessToken
+
+    /**
+     * Constructor for the class. It initializes the session, endpoint, and access token.
+     */
+    TowerFusionEnv() {
+        this.session = Global.session as Session
+        final towerConfig = session.config.navigate('tower') as Map ?: [:]
+        final env = SysEnv.get()
+        this.endpoint = endpoint0(towerConfig, env)
+        this.accessToken = accessToken0(towerConfig, env)
+    }
+
+    /**
+     * Return any environment variables relevant to Fusion execution. This method is called
+     * by {@link nextflow.fusion.FusionEnvProvider#getEnvironment} to determine which
+     * environment variables are needed for the current run.
+     *
+     * @param scheme The scheme for which the environment variables are needed (currently unused)
+     * @param config The Fusion configuration object
+     * @return A map of environment variables
+     */
+    @Override
+    Map<String, String> getEnvironment(String scheme, FusionConfig config) {
+
+        // TODO(amiranda): Hardcoded for now. We need to find out how to obtain
+        // the concrete product SKU and version. Candidate: FusionConfig?
+        final product = 'fusion'
+        final version = '2.4'
+
+        try {
+            final token = getLicenseToken(product, version)
+            return [
+                'FUSION_LICENSE_TOKEN': token,
+            ]
+        } catch (Exception e) {
+            log.warn("Error retrieving Fusion license information: ${e.message}")
+            return [:]
+        }
+    }
+
+    /**
+     * Send a request to Platform to obtain a license-scoped JWT for Fusion. The request is authenticated using the
+     * Platform access token provided in the configuration of the current session.
+     *
+     * @throws AbortOperationException if a Platform access token cannot be found
+     *
+     * @return The signed JWT token
+     */
+    protected String getLicenseToken(product, version) {
+        // FIXME(amiranda): Find out how to obtain the product and version
+        // Candidate: FusionConfig?
+
+        if (accessToken == null) {
+            throw new AbortOperationException("Missing personal access token -- Make sure there's a variable TOWER_ACCESS_TOKEN in your environment")
+        }
+
+        final req = HttpRequest.newBuilder()
+            .uri(URI.create("${endpoint}/${LICENSE_TOKEN_ENDPOINT}").normalize())
+            .header('Content-Type', 'application/json')
+            .header('Authorization', "Bearer ${accessToken}")
+            .POST(
+                HttpRequest.BodyPublishers.ofString(
+                    new Gson().toJson(
+                        new LicenseTokenRequest(
+                            product: product,
+                            version: version
+                        ),
+                        LicenseTokenRequest.class
+                    ),
```

Comment:
Built-in `JsonOutout.toJson(Map)` can be easier

Reply:
I did it this way for symmetry with the [deserialization](https://github.com/nextflow-io/nextflow/blob/e7b10823f5937cfcf32c76f165651300e6216166/plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFusionEnv.groovy#L149) but I don't have a preference for one or the other.

Reply:
I'm fine keeping gson, then it would be better to isolate into its own method